tests/unit/test_utils.py

```diff
@@ -95,3 +95,28 @@ def test_warn(self):
         assert warn_message in str(warning.message)
         assert __file__ in str(warning.message)
         assert warn_source == warning.source
+
+
+def test_transform_types_copies_data():
+    data = {"attr": "spam"}
+    new_data, _ = utils._transform_types(data, {})
+
+    assert new_data is not data
```

Comment:
```suggestion
    new_data, files = utils._transform_types(data, {})

    assert new_data is not data
    assert new_data == data
    assert files == {}
```

Reply:
Added this but renamed the tests a bit as this changes what is being tested.